src/test/java/uk/gov/hmcts/ccd/domain/service/common/CaseAccessGroupUtilsTest.java

```diff
@@ -0,0 +1,546 @@
+
+        verify(caseDataService, times(2)).getDefaultSecurityClassifications(
+            eq(caseTypeDefinition),
+            caseDataCaptor.capture(),
+            any()
+        );
+        // must have CaseAccessGroups value
+        assertNotEquals(data.get(CaseAccessGroupUtils.CASE_ACCESS_GROUPS).size(),
+            caseDataCaptor.getValue().get(CaseAccessGroupUtils.CASE_ACCESS_GROUPS).size());
+    }
+
+    private CaseTypeDefinition createCaseTypeDefinitionWithoutCaseAccessGroup() {
+
+        CaseFieldDefinition caseFieldDefinition = new CaseFieldDefinition();
+        caseFieldDefinition.setId("myTextField");
+        caseFieldDefinition.setFieldTypeDefinition(createFieldTypeDefinition(TEXT));
+
+        List<CaseFieldDefinition> caseFields = new ArrayList<>();
+        caseFields.add(caseFieldDefinition);
+
+        var newCaseTypeDefinition = new CaseTypeDefinition();
+        newCaseTypeDefinition.setCaseFieldDefinitions(caseFields);
+
+        return newCaseTypeDefinition;
+    }
+
+    private CaseTypeDefinition createCaseTypeDefinitionWithCaseAccessGroup() {
+        var newCaseTypeDefinition = createCaseTypeDefinitionWithoutCaseAccessGroup();
+
+        CaseFieldDefinition caseAccessGroupsFieldDefinition = new CaseFieldDefinition();
+        caseAccessGroupsFieldDefinition.setId(CaseAccessGroupUtils.CASE_ACCESS_GROUPS);
+        caseAccessGroupsFieldDefinition
+            .setFieldTypeDefinition(createFieldTypeDefinition(CaseAccessGroupUtils.CASE_ACCESS_GROUPS));
+
+        newCaseTypeDefinition.getCaseFieldDefinitions().add(caseAccessGroupsFieldDefinition);
+        return newCaseTypeDefinition;
+    }
+
+    private FieldTypeDefinition createFieldTypeDefinition(String type) {
+        FieldTypeDefinition fieldTypeDefinition = new FieldTypeDefinition();
+        fieldTypeDefinition.setId(type);
+        fieldTypeDefinition.setType(type);
+        return fieldTypeDefinition;
+    }
+
+    private Map<String, JsonNode> getClonedCaseAccessGroups(Map<String, JsonNode> data) {
+        Map<String, JsonNode> clonedData = new HashMap<>(data);
+
+        if (data.containsKey(CaseAccessGroupUtils.CASE_ACCESS_GROUPS)) {
+            var expectedCaseAccessGroupsJson = data.get(CaseAccessGroupUtils.CASE_ACCESS_GROUPS).deepCopy();
+            clonedData.put(CaseAccessGroupUtils.CASE_ACCESS_GROUPS, expectedCaseAccessGroupsJson);
+        }
+
+        return clonedData;
+    }
+
+    private void mockGetDefaultSecurityClassificationsResponse(CaseDetails caseDetails) {
+        JsonNode caseAccessGroupJsonNode = caseDetails.getData().get(CaseAccessGroupUtils.CASE_ACCESS_GROUPS);
+        Map<String, JsonNode> dataClassification = new HashMap<>();
+        dataClassification.put(CaseAccessGroupUtils.CASE_ACCESS_GROUPS, caseAccessGroupJsonNode);
+
+        doReturn(dataClassification).when(caseDataService).getDefaultSecurityClassifications(
```

Comment:
I don't think we should be mocking the response back from this method.  I expected us to run this method, get a value and verify it is the value we expect it to be?

Reply:
As I was not getting the results back I used the piece of code from TimeToLiveServcieTest on line 311.